### PR TITLE
(GH-53) Add xml parser filter for control characters

### LINKED ISSUE #53: Filter control chars from log file content
Log file content may contain control characters which should be filtered out.

### PR DESCRIPTION
**Pull request by MartinEgli**
The MSBuild.log may contain NUL NUL values. This is caused by bad logger. I would also like to have the logger corrected. The XML parser should nevertheless filter wrong char.

Fixes #53

### CODE REVIEW

**Comment by pascalberger** [comment]
@MartinEgli Thanks for your contribution. I would like to have this also covered by a test case. Are you in a position to add a test case or if not provide an (anonymized) file which can be used to reproduce this issue?

**Comment by MartinEgli** [comment]
Add TestData and Unittest.
Sorry for geman text.

**Comment by MartinEgli** [comment]
Update packages

File: src/Cake.Issues.MsBuild.Tests/Cake.Issues.MsBuild.Tests.csproj [hunk 1]
```diff
@@ -32,24 +33,28 @@
...
 
   <ItemGroup>
-    <PackageReference Include="Microsoft.NET.Test.Sdk" Version="15.8.0" />
-    <PackageReference Include="Cake.Core" Version="0.28.0" />
-    <PackageReference Include="Cake.Testing" Version="0.28.0" />
-    <PackageReference Include="Cake.Issues" Version="0.6.0" />
-    <PackageReference Include="Cake.Issues.Testing" Version="0.6.0" />
-    <PackageReference Include="Shouldly" Version="3.0.0" />
+    <PackageReference Include="Microsoft.NET.Test.Sdk" Version="15.9.0" />
+    <PackageReference Include="Cake.Core" Version="0.30.0" />
```
Comment (pascalberger) [inline src/Cake.Issues.MsBuild.Tests/Cake.Issues.MsBuild.Tests.csproj]: There's no need to update this dependency as it should link to the lowest version without breaking change for maximum compatibility. Same for other dependencies (the failing build is also caused by the Cake.Issues update)

Comment (MartinEgli) [reply src/Cake.Issues.MsBuild.Tests/Cake.Issues.MsBuild.Tests.csproj]: I agree.

**Comment by MartinEgli** [comment]
I agree.

**Comment by pascalberger** [comment]
@MartinEgli your changes have been merged, thanks for your contribution 👍

**Pull request merged**